src/cursor.rs

```diff
@@ -279,7 +279,7 @@ impl Cursor {
     ) -> Result<Cursor> {
 
         let mut stream = stream;
-        let mut socket = stream.get_socket();
+        let socket = stream.get_socket();
```

Comment:
This seems unrelated;  did the compiler give a warning here or something?

Reply:
yeah, this was a warning that i noticed in the travis build; figured I'd fix it while I'm here, although I wonder why this wasn't noticed or warned about previously

Reply:
From testing it locally, this warning (and the one regarding the doc comment you fixed) only occur on 1.21, not 1.20. It's unclear to me whether this is due to mutability no longer being required with the new compiler/standard library or if the compiler merely improved its existing ability to detect unnecessary mutability qualifiers.

Reply:
Since the crate builds and the tests pass with 1.20, I'm going to leave this change 👋